### PR TITLE
Fixes #993: Made cimtype() and type_from_name() conversion functions part of public API

### LINKED ISSUE #993: Make type_from_name() and cimtype() part of the API
This issue was created to resolve this TODO in _server.py:

W:1014, 0: TODO: We should make type_from_name() and cimtype() part of the API (fixme)

These functions are defined in cim_types.py, and _server.py being a high level user of the core API should not have to use internal functions.

We agreed that these functions should be made part of the public API.

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see commit message.
Ready for review and merge.

### CODE REVIEW

File: docs/changes.rst [hunk 1]
```diff
@@ -225,6 +225,14 @@ Enhancements
   and to `tocimxmlstr()` to allow control over whether the object is
   interpreted as a value or as a declaration. (Issue #950).
 
+* Added a new conversion function: `cimtype()` takes a CIM data typed value
```
Comment (KSchopmeyer) [inline docs/changes.rst]: shouldn't we say Added a new conversion function to the public API?

File: pywbem/cim_types.py [hunk 2]
```diff
@@ -872,33 +875,33 @@ def type_from_name(type_name):
...
     :class:`~pywbem.Uint8`.
 
     For CIM data type names ``"string"`` and ``"char16"``, the
     :term:`unicode string` type is returned (Unicode strings are the preferred
     representation for these CIM data types).
 
-    The returned type can be used to 'cast' a value to that type in most cases,
-    except for "string" and "char16" typed values:
+    The returned type can be used as a constructor from a differently typed
+    input value, in many cases. Notable limitations are:
```
Comment (KSchopmeyer) [inline pywbem/cim_types.py]: Remove the comma  value, in. M change many to some.

File: docs/changes.rst (hunk 1 above)
Comment (KSchopmeyer) [reply docs/changes.rst]: That cements what we are doing in the first sentence.  Applies to the following also.

Comment (andy-maier) [reply docs/changes.rst]: DONE. Both changed as proposed.

File: pywbem/cim_types.py (hunk 2 above)
Comment (andy-maier) [reply pywbem/cim_types.py]: Comma: DONE.
"in many cases": They are really many cases for which the statement is true: All CIM integer types from an integer; all CIM real types from integer and float; CIMDateTime from Python datetime and timedelta and string in datetime format; CIM string and char16 from byte string except for the limitation, CIM reference from string in WBEM URI format and from CIMInstanceName.

**Comment by andy-maier** [comment]
Resolved the comments as stated, and considering this one approved since you considered these minor changes.

**Pull request merged**